### PR TITLE
feat: implement and test dss start

### LINKED ISSUE #64: Implement DSS `start` command
### Why it needs to get done

The start command starts a specified notebook within the DSS environment. This command is expect to start a stopped notebook

```
dss start my-notebook
```

Please use the UX [spec](https://docs.google.com/document/d/1v0wU8TBb-ui6RcjGRbLwA6XFZnWbuiVb2TPmEgwKejY/edit#heading=h.rfcm5y81sk1z) documentation .

### What needs to get done

1. Implement start command
2. Implement integration test with Microk8s

### When is the task considered done

1. Command is implemented
2. Tests are passing

### PR DESCRIPTION
**Pull request by NohaIhab**
closes #64

## Summary
* Implement `dss start` command
* Add unit tests for `dss start` success and failure cases
* Implement the logging and exit code for `dss start` to comply with #78 

## Testing
```
git checkout origin/kf-5522-implement-start-notebook
pip install -e .
dss initialize --kubeconfig ~/.kube/config
dss create test-notebook --kubeconfig ~/.kube/config
dss stop test-notebook --kubeconfig ~/.kube/config
dss list --kubeconfig $KUBECONFIG --wide
```
You should see:
```
2024-04-22 19:09:23 [INFO] [list] [list_notebooks]: 
 Name           Image                                     URL       
 test-notebook  kubeflownotebookswg/jupyter-scipy:v1.8.0  (Stopped)
```
Then start the notebook:
```
dss start test-notebook --kubeconfig ~/.kube/config
dss list --kubeconfig $KUBECONFIG --wide
```
You should see:
```
2024-04-22 19:10:47 [INFO] [list] [list_notebooks]: 
 Name           Image                                     URL                      
 test-notebook  kubeflownotebookswg/jupyter-scipy:v1.8.0  http://10.152.183.155:80
```

### CODE REVIEW

File: src/dss/main.py [hunk 1]
```diff
@@ -155,6 +156,35 @@ def stop_notebook_command(kubeconfig: str, notebook_name: str):
...
+    "--kubeconfig",
+    help=f"Path to a Kubernetes config file. Defaults to the value of the KUBECONFIG environment variable, else to '{KUBECONFIG_DEFAULT}'.",  # noqa E501
+)
+def start_notebook_command(name: str, kubeconfig: str):
+    """
+    Starts a stopped notebook in the DSS environment.
+    \b
+    Example:
+        dss start my-notebook
+    """
```
Comment (DnPlas) [inline src/dss/main.py:176]: I'm pretty sure you are following a standard, but printing the help for this yields the following:

```
$ dss start --help
Usage: dss start [OPTIONS] NAME

  Starts a stopped notebook in the DSS environment. Example:     dss start
  my-notebook

Options:
  --kubeconfig TEXT  Path to a Kubernetes config file. Defaults to the value
                     of the KUBECONFIG environment variable, else to
                     './kubeconfig'.
  --help             Show this message and exit.
```

Should we ensure the formatting is okay?

File: src/dss/start.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    """
+
+    if not does_notebook_exist(
+        name=name, namespace=DSS_NAMESPACE, lightkube_client=lightkube_client
+    ):
+        logger.error(f"Failed to start Notebook. Notebook {name} does not exist.")
+        logger.info("Run 'dss list' to check all notebooks.")
+        raise RuntimeError(f"Failed to start Notebook. Notebook {name} does not exist.")
+
+    obj = Deployment.Scale(
```
Comment (DnPlas) [inline src/dss/start.py:31]: This, alongside with `dss stop` can affect any `Deployment` in the `DSS_NAMESPACE`. This behaviour is not desired, we would like to only start/stops Notebook resources. The way you can test this is to try to stop `mlflow` after initialising dss `dss stop mlflow  --kubeconfig $KUBECONFIG` and checking the `Deployment`:

```
$ kubectl get deployments -A | grep mlflow
dss               mlflow                    1/1     1            1           10m

```

Then you can start `mlflow` w/o issues:

```
$ dss start mlflow  --kubeconfig $KUBECONFIG
2024-04-19 15:09:26 [INFO] [main] [start_notebook_command]: Executing start command
2024-04-19 15:09:26 [INFO] [start] [start_notebook]: Starting the notebook mlflow. Check `dss list` for the status of the notebook.
```

**Review by DnPlas — COMMENTED** [review]
Thanks @NohaIhab, left a few comments, it looks good overall, I understand we are following the same pattern and style for other commands, so nothing worrying about that. The only thing I would consider worrying is that this command can affect any `Deployment` in the `dss` namespace. I have left a comment, let's discuss.

File: src/dss/start.py (hunk 2 above)
Comment (NohaIhab) [reply src/dss/start.py:31]: this is a very good point, and I think we will have to change it in this command and all the other commands affected (`remove`, `stop` and possibly `logs`).
I'm filing an issue to track this and tackle it separately.

Comment (NohaIhab) [reply src/dss/start.py:32]: #84

File: src/dss/main.py (hunk 1 above)
Comment (NohaIhab) [reply src/dss/main.py:208]: fixed it

Comment (NohaIhab) [reply src/dss/main.py:208]:
```
dss stop --help
Usage: dss stop [OPTIONS] NOTEBOOK_NAME

  Stops a running notebook in the DSS environment.

  Example:
      dss stop my-notebook

Options:
  --kubeconfig TEXT  Path to a Kubernetes config file. Defaults to the value
                     of the KUBECONFIG environment variable, else to
                     './kubeconfig'.
  --help             Show this message and exit.
```

File: src/dss/start.py (hunk 2 above)
Comment (DnPlas) [reply src/dss/start.py:32]: Okay, as long as we come back to this.

**Review by misohu — APPROVED** [review]
Good Job Noha :)

**Pull request merged**